### DISCUSSION TITLE
When using ```torch.no_grad()```, will deepspeed *not* perform memory optimization on some parameters of ```no_grad```?

### DISCUSSION THREAD

**Discussion by ojipadeson** [deepspeedai/DeepSpeed#5805] [Q&A]
I froze part of the LLM model and introduced it into my custom model. I found that in this case the memory usage of the first GPU is always much higher than the others. Does it mean that deepspeed will not perform memory optimization for the non-gradient part.

**Comment by tjruwase** [comment]
@ojipadeson, DeepSpeed does optimize for parameters that have no gradients, i.e., frozen parameters. See examples below. 

https://github.com/microsoft/DeepSpeed/blob/ffe0af23575c4f03a07408eacfc50b1a58781429/deepspeed/runtime/zero/stage_1_and_2.py#L310-L312

https://github.com/microsoft/DeepSpeed/blob/ffe0af23575c4f03a07408eacfc50b1a58781429/deepspeed/runtime/zero/stage3.py#L446

Can you please provide repro steps to help investigate what you are observing? Thanks!
